src/main/java/htmlpublisher/HtmlPublisherTarget.java

```diff
@@ -202,17 +212,26 @@ protected String getTitle() {
         protected File dir() {
             return getBuildArchiveDir(this.build);
         }
+        
+        /**
+         * Gets {@link HtmlPublisherTarget}, for which the action has been created.
+         * @return HTML Report description
+         * @since TODO
+         */
+        public @Nonnull HtmlPublisherTarget getHTMLTarget() {
+            return HtmlPublisherTarget.this;
+        }
     }
 
-    public void handleAction(AbstractBuild<?, ?> build) {
+    public void handleAction(Run<?, ?> build) {
         // Add build action, if coverage is recorded for each build
         if (this.keepAll) {
             build.addAction(new HTMLBuildAction(build, this));
         }
     }
 
-    public Action getProjectAction(AbstractProject project) {
-        return new HTMLAction(project, this);
+    public Action getProjectAction(AbstractItem item) {
```

Comment:
Definitely :)
BTW, I see no problem in the usage of the Job's parent class till Actions begin depending on Job APIs (which we would like to avoid)